contracts/farm-bot.sol

```diff
@@ -137,11 +146,29 @@ contract FarmBot is Owned, FarmbotERC20 {
     }
 
     // convenience method for anyone considering calling claimRewards (who may want to compare bounty to gas cost)
-    function previewBounty() external view returns (TokenAmount[] memory) {
-        uint _leftoverBalance = rewardsToken.balanceOf(address(this));
-        uint _interestEarned = stakingRewards.earned(address(this));
-        TokenAmount[] memory _rewardsTokenBalances = new TokenAmount[](1);
-        _rewardsTokenBalances[0] = TokenAmount(rewardsToken, _interestEarned + _leftoverBalance);
+    // Annoyingly, the MoolaStakingRewards.earnedExternal method is not declared as a view, so we cannot declare this
+    // method as a view itself.
+    function previewBounty() external returns (TokenAmount[] memory) {
```

Comment:
This is annoying... but not sure how to work around this.

Reply:
does that mean it will cost gas now to call this? not great..

Reply:
Not necessarily? I think you can do an e.g., `farmBotContract.methods.previewBounty().call()` using web3 which will emulate the call locally which should be fine? Although not really sure to be honest.